### PR TITLE
Godel Phase 2 Code

### PR DESCRIPTION
**Pull request by Jmeyer1292**
First off, my apologies for what is an enormous pull request. I fully expect this to get torn apart and put into more managable components shortly, but I promised to get the code out here ASAP.

The major highlights are:
1. Kinect 2 integration
2. Calibration routines 
3. Laser scan analysis code
4. Primitive cartesian motion using Descartes

Thanks to everyone involved for their support and guidance in creating this software.

I used https://github.com/code-iai/iai_kinect2 to communicate with the Kinect 2. The trajectory planner uses a feature branch (velocity pruning) of descartes which has not yet been accepted. I'll keep updating this post as we refine this PR.

Thank you

### CODE REVIEW

File: godel_path_planning/include/godel_path_planning/trajectory_cartesian_motion.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#ifndef TRAJECTORY_CARTESIAN_MOTION_H
```
Comment (jrgnicho) [inline godel_path_planning/include/godel_path_planning/trajectory_cartesian_motion.h:1]: Is this file needed?

**Comment by Jmeyer1292** [comment]
Closed as a result of the many recent PRs #64 through #81.
